apm-collector/apm-collector-worker/src/main/java/org/skywalking/apm/collector/worker/httpserver/AbstractPost.java

```diff
@@ -1,74 +1,74 @@
+ * handle a POST request.
+ *
+ * <p>verride the {@link #onReceive(Map, JsonObject)} method to support a search service.
+ *
  * @author pengys5
+ * @since v3.0-2017
  */
-
-public abstract class AbstractPost extends AbstractLocalAsyncWorker {
+public abstract class AbstractPost extends AbstractServlet {
 
     public AbstractPost(Role role, ClusterWorkerContext clusterContext, LocalWorkerContext selfContext) {
         super(role, clusterContext, selfContext);
     }
 
-    @Override
-    final public void onWork(Object message) throws Exception {
-        onReceive(message);
+    /**
+     * Add final modifier to avoid the subclass override this method.
+     *
+     * @param parameter {@link Object} data structure of the map
+     * @param response {@link Object} is a out parameter
+     * @throws Exception
+     */
+    @Override final protected void onWork(Object parameter, Object response) throws Exception {
+        super.onWork(parameter, response);
     }
 
-    protected abstract void onReceive(Object message) throws Exception;
-
-    static class PostWithHttpServlet extends AbstractHttpServlet {
+    static class PostWithHttpServlet extends HttpServlet {
 
         private Logger logger = LogManager.getFormatterLogger(PostWithHttpServlet.class);
 
-        private final LocalAsyncWorkerRef ownerWorkerRef;
+        private final LocalSyncWorkerRef ownerWorkerRef;
 
-        PostWithHttpServlet(LocalAsyncWorkerRef ownerWorkerRef) {
+        PostWithHttpServlet(LocalSyncWorkerRef ownerWorkerRef) {
             this.ownerWorkerRef = ownerWorkerRef;
         }
 
-        @Override
-        final protected void doPost(HttpServletRequest request,
-                                    HttpServletResponse response) throws ServletException, IOException {
-            JsonObject resJson = new JsonObject();
+        /**
+         * Override the {@link HttpServlet#doPost(HttpServletRequest, HttpServletResponse)} method, receive the
+         * parameter from request then send parameter to the owner worker.
+         *
+         * @param request an {@link HttpServletRequest} object that contains the request the client has made of the
+         * servlet
+         * @param response {@link HttpServletResponse} object that contains the response the servlet sends to the
+         * client
+         * @throws ServletException if the request for the POST could not be handled
+         * @throws IOException if an input or output error is detected when the servlet handles the request
+         */
+        @Override final protected void doPost(HttpServletRequest request,
+            HttpServletResponse response) throws ServletException, IOException {
             try {
-                BufferedReader bufferedReader = request.getReader();
-                streamReader(bufferedReader);
-                reply(response, resJson, HttpServletResponse.SC_OK);
+                Map<String, String[]> parameter = request.getParameterMap();
```

Comment:
Should we process the request body context ? because `Request#getParameterMap()` only fetch parameters are contained in the query string or posted form data.